### PR TITLE
self grading feature for the UI

### PR DESCRIPTION
**Pull request by ruhansharief22**
Self grading feature:
- Changes to the ui-landing docker images and its contents - html, js, css, python,
- minor changes to the local grading script in the login container
- changes in docker compose file - volume mounts and image versions

### CODE REVIEW

File: nested-labvm/atd-docker/docker-compose.yml [hunk 1]
```diff
@@ -24,7 +24,7 @@ services:
       - /home/arista/arista-dir:/home/arista:rw
   login:
     container_name: atd-login
-    image: us.gcr.io/atd-testdrivetraining-dev/atddocker_login:1.0.11
+    image: us.gcr.io/atd-testdrivetraining-dev/atddocker_login:selfgrading
```
Comment (ruhansharief22) [inline nested-labvm/atd-docker/docker-compose.yml]: Ruhan: Need to change this to a deployable tag

File: nested-labvm/atd-docker/docker-compose.yml [hunk 2]
```diff
@@ -85,11 +86,12 @@ services:
       - /opt/ceos:/opt/ceos:rw
   uilanding:
     container_name: atd-uilanding
-    image: us.gcr.io/atd-testdrivetraining-dev/atddocker_uilanding:1.0.7
+    image: us.gcr.io/atd-testdrivetraining-dev/atddocker_uilanding:selfgrading
```
Comment (ruhansharief22) [inline nested-labvm/atd-docker/docker-compose.yml]: Need to change this to a deployable tag

File: nested-labvm/atd-docker/docker-compose.yml [hunk 3]
```diff
@@ -194,6 +196,14 @@ services:
       - "/home/arista/arista-dir/apps/webui:/config:rw"
     networks:
       - atd_nginx
+  grader:
+    container_name: atd-grader
+    image: us.gcr.io/atd-testdrivetraining-dev/atddocker_selfgrading:latest
```
Comment (ruhansharief22) [inline nested-labvm/atd-docker/docker-compose.yml]: Need to change this to a deployable tag

File: nested-labvm/atd-docker/uilanding/src/html/js/backend.js [hunk 4]
```diff
@@ -43,7 +43,8 @@ $('#labGrading').click(function (event) {
     document.getElementById('mainContent').style.display = 'none'
     document.getElementById('labStatusContent').style.display = 'none'
     document.getElementById('labGradingData').style.display = 'block'
-    clearInterval(labStatusInterval)
+    loadData("labGrading")
+    //clearInterval(labStatusInterval)
```
Comment (ruhansharief22) [inline nested-labvm/atd-docker/uilanding/src/html/js/backend.js]: this should be uncommented, provided there are relative changes made to overcome the 'labStatusInterval undefined' error

File: nested-labvm/atd-docker/uilanding/src/html/js/backend.js [hunk 5]
```diff
@@ -106,16 +107,166 @@ function getLabStatus() {
...
+      })
+      .then(data => {
+        loadGradingData(data.grading)
+        document.getElementById('grade-button').disabled = false;
+        displayConvertTime(data.timestamp)
+      })
+      .catch(error => {
+        displayGradeError("No details to show")
+        document.getElementById('grade-button').disabled = false;
+        displayConvertTime(data.timestamp)
```
Comment (ruhansharief22) [inline nested-labvm/atd-docker/uilanding/src/html/js/backend.js]: this line fails. should be commented

File: nested-labvm/atd-docker/uilanding/src/uilanding.py [hunk 6]
```diff
@@ -349,8 +352,105 @@ def get(self):
         docker_conn= docker.from_env()
         login_container = docker_conn.containers.get('atd-login')
         login_container.exec_run(f'sudo python3 /usr/local/bin/resetVMs.py')
+class LabGradingHandler(BaseHandler):
+    _FILE_PATH = "/etc/opt/graderlogs/"
+    _FILE_PATTERN = "*-output.json"
+    _TITLE_MAPPINGS = {
```
Comment (ruhansharief22) [inline nested-labvm/atd-docker/uilanding/src/uilanding.py:358]: check on these mapping and accordingly change the template file name

File: nested-labvm/atd-docker/uilanding/src/uilanding.py [hunk 7]
```diff
@@ -349,8 +352,105 @@ def get(self):
...
+        """
+        client = docker.from_env()
+        # Call for the running configs
+        login_container = client.containers.get('atd-login')
+        login_container.exec_run(f'sudo localGrading.py')
+        #image_name = client.images.pull("selgrading:latest",no_cache=True)
+        #import pdb;pdb.set_trace()
+        #image_name = client.images.pull("gcr.io/atd-testdrivetraining-dev/selfgrading:1.0.0")
+        #image_name = "gcr.io/atd-testdrivetraining-dev/selfgrading:1.0.0"
+        image_name = "us.gcr.io/atd-testdrivetraining-dev/atddocker_selfgrading:latest"
```
Comment (ruhansharief22) [inline nested-labvm/atd-docker/uilanding/src/uilanding.py]: change the image tag and clear up the commented lines

File: nested-labvm/atd-docker/uilanding/src/uilanding.py (hunk 6 above)
Comment (ruhansharief22) [reply nested-labvm/atd-docker/uilanding/src/uilanding.py:358]: changed the template name

File: nested-labvm/atd-docker/uilanding/src/uilanding.py (hunk 7 above)
Comment (ruhansharief22) [reply nested-labvm/atd-docker/uilanding/src/uilanding.py]: removed the commented lines. tag to be updated

File: nested-labvm/atd-docker/docker-compose.yml (hunk 3 above)
Comment (rparkinarista) [reply nested-labvm/atd-docker/docker-compose.yml]: We do need a method to handle feature tags like this, I have one for live chat.

**Review by ruhansharief22 — COMMENTED** [review]
Reviewed

**Pull request merged**